### DISCUSSION TITLE
Scenario level parallelism

### DISCUSSION THREAD

**Discussion by ThomasHeijtink** [reqnroll/Reqnroll#53] [Ideas]
I know this has been discussed a lot in specflow discussions. But since I don't see a discussion yet regarding this topic I thought I just drop this here as a placeholder for discussing scenario level parallelism. 

Since the primary resolution of BDD are user/business requirements it often means that tests cover more than just simpel (high resolution) classes . But also cover web api's, message bus and databases (to name a few). This has a significant impact on the performance and therefore the feedback cycle. Scenario level parallelism would greatly aid in decreasing the feedback cycle duration.

Although it will probably also tie into some of the work that would go into #11. I still wanted to mention this as a separate "idea". The most important thing is that when #11 is beyond the POC stage that it would be a great opportunity to keep scenario level parallelism in mind.

**Comment by ajeckmans** [comment]
I think the work on #16 will be relevant for this to work completely without gotchas as well.

**Reply by ThomasHeijtink** [reply]
Yes that's true. For our parallelism we've copied some of the table helper code and rolled our own value retrieval functionality. We mainly use it to convert strings to id's (in our case GUID's). For all other value retrieval, static code has been sufficient for us. Maybe we could exchange notes in the ValueRetriever discussion if you're interested.

**Comment by gasparnagy** [comment]
@ThomasHeijtink Thx. Yes, I agree that this is important. I would see #40 as a prerequisite of this and we would need to have a plan as well, how we could address this.

In terms of focusing of our efforts, #11 is related, but I think the concrete technical solution is independent of #11. I more think that it will finally route to better paralel handling in BoDi - but this is just a feeling.

**Comment by gasparnagy** [comment]
Duplicate topic: https://github.com/orgs/reqnroll/discussions/117

I close this
